src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -881,6 +894,24 @@ public CharSeq takeWhile(Predicate<? super Character> predicate) {
     }
 
     @Override
+    public <T1, T2, T3> Tuple3<Vector<T1>, Vector<T2>, Vector<T3>> unzip3(
+            Function<? super Character, Tuple3<? extends T1, ? extends T2, ? extends T3>> unzipper) {
+        Objects.requireNonNull(unzipper, "unzipper is null");
+        Vector<T1> xs = Vector.empty();
+        Vector<T2> ys = Vector.empty();
+		Vector<T3> zs = Vector.empty();
+        for (int i = 0; i < length(); i++) {
+            final Tuple3<? extends T1, ? extends T2, ? extends T3> t = unzipper.apply(back.charAt(i));
+            xs = xs.append(t._1);
+            ys = ys.append(t._2);
+			zs = zs.append(t._3);
+        }
+        return Tuple.of(xs.isEmpty() ? Vector.<T1> empty() : xs, 
+						ys.isEmpty() ? Vector.<T2> empty() : ys,
+						zs.isEmpty() ? Vector.<T3> empty() : zs);
```

Comment:
Hi. I pretty much copied the unzip method, and that one actually had xs.length() == 0, but anyway, you're right, now that I look at it with my brains on the isEmpty makes no sense. :)